blog.md

```diff
@@ -0,0 +1,203 @@
+Base.:+(a::Tropical, b::Tropical) = Tropical(max(a.n, b.n))
+```
+Then users can simply use the tropical algebra, for example
+```julia
+julia> using TropicalNumbers
+
+julia> a = Tropical(1.0)
+1.0ₜ
+
+julia> b = Tropical(2.0)
+2.0ₜ
+
+julia> a + b
+2.0ₜ
+
+julia> a * b
+3.0ₜ
+```
+and opeartions of vectors and matrices also work
+```julia
+julia> A = Tropical.(rand(2, 2))
+2×2 Matrix{Tropical{Float64}}:
+ 0.2238665251106623ₜ  0.18355043791779635ₜ
+ 0.3673107532619566ₜ   0.1573950170887196ₜ
+
+julia> B = Tropical.(rand(2))
+2-element Vector{Tropical{Float64}}:
+ 0.16479545470285972ₜ
+  0.3666513822212566ₜ
+
+julia> A * B
+2-element Vector{Tropical{Float64}}:
+ 0.5502018201390529ₜ
+ 0.5321062079648163ₜ
+```
+Since we define the tropical number as a subtype of `Number`,
+```julia
+julia> isbitstype(Tropical{Float64})
+true
+```
+which means the storage of tropical number in the memory is continium.
+
+All these things work naturely, users will be able to use the tropical algebra just like real numbers.
+
+### High performance CUDA kernel
+
+For the lower level kernel, due to the requirement of performance, we choose to use `C-Cuda` directly and wrap the code with a Julia interface.
+In this part, we simply used some skills common in GEMM kernels to speed up the code, and we learned a lot from the repo [CUDA_gemm](https://github.com/Cjkkkk/CUDA_gemm) by [Cjkkk](https://cjkkkk.github.io).
+Here we will briefly introduce these skills, and for more detailed introduction, I recommand this [blog](https://zhuanlan.zhihu.com/p/441146275) in Chinese.
+
+As we all know, GPU is fast because it have a large amount of cores, and the cores are grouped into blocks.
+In the GPU kernels, we can allocate tasks to different blocks by decomposing them into a series of unrelated sub-tasks. Furthermore, we can further decompose the sub-tasks and assign them to threads for execution.
+However, this large number of parallel tasks also leads to another issue, which is the significant data payload. 
+Although the shared memory accessible by blocks is similar in speed to the L2 cache in CPUs, its capacity is limited. 
+Therefore, it is not possible to load all data into shared memory at once, and only the data required for the current block calculation can be loaded each time.
+Unfortunately, GEMM is a very memory-intensive operation, for example, when calculating the GEMM between a $M \times K$ matrix and a $K \times N$ matrix, if we use the naive way, i.e. evaluate the element in the result matrix one by one, we will have to load $M \times K \times N$ elements from the slow global memeory directly to registers in the whole process, and this generally far exceeds the data bandwidth of the GPU, resulting in severe performance issues.
+
+To avoid the heavy data loading, we first split the target matrix into blocks with size $BM \times BN$, and each GPU block will be used to calculate one of the tiled matrix, as shown in the fig below: 
```

Comment:
missing key word: tiling

Reply:
`tiling` is still missing. I think this keyword can help readers find more contents related to optimizing matrix multiplication. Try to mention it somewhere?

Reply:
You should add a sentense like "We use the tiling technique to ..."